.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,10 +1,24 @@
-###### Before submitting the PR, please make sure you do the following
+Closes https://github.com/metabase/metabase/pull/[issue number]
 
-- [ ] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.
+### Description
 
-### Tests
+Describe the overall approach and the problem being solved.
+
+### How to verify
+
+Describe the steps to verify that the changes are working as expected.
+
+1. New question -> Sample Dataset -> ...
+2. ...
+
+#### Before & After Screenshots
+
+_Insert images here if sensible or remove the section_
```

Comment:
Warning: being ridiculously nitpicky. We don't always have the "before" state and some people also record videos. Should we consider using a more neutral title like "Demo"? We can extend the description to something like "Upload a demo video or before/after screenshots if sensible or remove the section"

Reply:
My 2 cents, I sometimes even need to run 2 versions at the same time to compare the result from the PR, I almost always want to do this, but it takes too much effort sometimes. It'd be great if the author has done this, as it will only take them some extra effort to save screenshots on the change they're editing. But that's much harder for the reviewers.

I know stats could help in some cases, but things like admin/settings or more destructive actions should better be tested locally still.

Reply:
@WiNloSt it is hard to generalize it as a useful practice since with many features that add non-existent before things it may be awkward. I added the nice wording Anton suggested